src/Features/CSharp/Portable/MakeMethodAsynchronous/CSharpMakeMethodAsynchronousCodeFixProvider.cs

```diff
@@ -116,6 +117,18 @@ TypeSyntax MakeGenericType(string type, ITypeSymbol typeArgumentFrom)
             }
         }
 
+        private static bool IsIterator(IMethodSymbol x)
+        {
+            return x.Locations
+                .Any(l => ContainsYield(l));
+
+            bool ContainsYield(Location l)
+                => l.FindNode(default).DescendantNodes().Any(n => IsYield(n));
```

Comment:
This will fire incorrectly for yields inside local functions.

Reply:
Consider just adding a recursive search and bailing out if we find a local function or an anonymous function.

Reply:
or using `node.IsReturnableConstruct()`

Reply:
Good catch. Thanks!